### PR TITLE
[Core][OTel] Pass schema URL into tracer

### PR DESCRIPTION
**Pull request by pvaneck**
This provides more information as part of the tracer's instrumentation scope.

Also, fixed an attribute name in the rename dict (`messaging_bus.destination` should be `message_bus.destination` as that is what is actually being used by the amqp libraries.

Part of #25838

### CODE REVIEW

File: sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/_schema.py [hunk 1]
```diff
@@ -17,9 +17,7 @@ class OpenTelemetrySchemaVersion(
 
 class OpenTelemetrySchema:
 
-    SUPPORTED_VERSIONS = [
-        OpenTelemetrySchemaVersion.V1_19_0,
-    ]
+    SUPPORTED_VERSIONS = (OpenTelemetrySchemaVersion.V1_19_0,)
```
Comment (xiangyan99) [inline sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/_schema.py:20]: Why we change the type?

Comment (pvaneck) [reply sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/_schema.py:20]: I remember hitting some mypy issues while using a list when I was testing things a while back. It doesn't seem to be a problem anymore, as I can't seem to recreate the issue. Would you prefer I keep this as a list even though we don't expect it to be mutated?

File: sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/_schema.py [hunk 2]
```diff
@@ -28,7 +26,7 @@ class OpenTelemetrySchema:
             "x-ms-client-request-id": "az.client_request_id",
             "x-ms-request-id": "az.service_request_id",
             "http.user_agent": "user_agent.original",
-            "messaging_bus.destination": "messaging.destination.name",
+            "message_bus.destination": "messaging.destination.name",
```
Comment (kashifkhan) [inline sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/_schema.py:29]: will this cause issues for folks using `messaging_bus` previously ?

Comment (pvaneck) [reply sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/_schema.py:29]: Nah, it shouldn't. The addition of the `OpenTelemetrySchema` class is still unreleased. And `messaging_bus.destination` was mainly just a typo as the intent for this dict is to remap keys used in the SDKs to converge on OTel defined keys. And nothing uses `messaging_bus.destination` (`message_bus.destination` is used however).

**Review by kashifkhan — APPROVED** [review]

**Pull request merged**